editor/utils/heartbeat.js

```diff
@@ -0,0 +1,153 @@
+import store from '../store';
+import {
+	getEditedPostTitle,
+	getEditedPostExcerpt,
+	getEditedPostContent,
+	getCurrentPostId,
+	isPostAutosaveDirty,
+} from '../store/selectors';
+import {
+	toggleAutosave,
+	removeNotice,
+	toggleNetworkIsConnected,
+	showDisconnectionNotice,
+	resetAutosave,
+	updateAutosaveStatusMessage,
+} from '../store/actions';
+
+/**
+ * WordPress dependencies
+ */
+import { doAction } from '@wordpress/hooks';
+
+const DISCONNECTION_NOTICE_ID = 'DISCONNECTION_NOTICE_ID';
+
+/**
+ * Set up the heartbeat functionality for Gutenberg.
+ */
+export function setupHeartbeat() {
+	const $document = jQuery( document );
+
+	/**
+	 * Configure heartbeat to refresh the wp-api nonce, keeping the editor authorization intact.
+	 *
+	 * @todo update the _wpnonce used by autosaves.
+	 */
+	$document.on( 'heartbeat-tick', ( event, response ) => {
+		if ( response[ 'rest-nonce' ] ) {
+			window.wpApiSettings.nonce = response[ 'rest-nonce' ];
+		}
+	} );
+
+	/**
+	 * Configure Heartbeat autosaves.
+	 */
+	const { dispatch, getState } = store;
+
+	/**
+	 * Autosave 'save' function that pulls content from Gutenberg state. Based on `wp.autosave.save`.
+	 *
+	 * @return {Object|boolean} postData The autosaved post data to send, or false if no autosave is needed.
+	 */
+	const save = function() {
+		// Bail early if autosaving is suspended or saving is blocked.
+		if ( wp.autosave.isSuspended || wp.autosave._blockSave ) {
```

Comment:
Where are these properties set?

Reply:
in autosave.js :) plugins can use the `wp.autosave` global to suspend and resume autosaving.